qiskit/providers/aer/backends/aerbackend.py

```diff
@@ -313,6 +322,217 @@ def _run(self, qobj, job_id=''):
 
         return Result.from_dict(output)
 
+    def _is_dynamic(self, circuit):
+        if not isinstance(circuit, QuantumCircuit):
+            return False
+        for inst, _, _ in circuit.data:
+            if isinstance(inst, (WhileLoopOp, ForLoopOp, IfElseOp, BreakLoopOp, ContinueLoopOp)):
+                return True
+        return False
+
+    def _decompose_subcircuits(self, circuit, basis_gates=None):
+        if circuit is None:
+            return None
+        if basis_gates is None:
+            basis_gates = self.configuration().basis_gates + ['for_loop', 'while_loop',
```

Comment:
I dont think any of these helper functions need `basis_gates=None`. They should have a required basis gates arg, and you should do this formatting of basis gates to add the loop functions in the call to the `_compile` function